app/controllers/application_controller.rb

```diff
@@ -51,6 +51,12 @@ def utm_parameters
     params.permit(:utm_source, :utm_medium, :utm_campaign, :utm_term, :utm_content)
   end
 
+protected
+
+  def after_sign_in_path_for(_resource)
```

Comment:
You've added this method, but I can't see where it's used.

Reply:
It's an implicit Devise thing - if it's present, it's used as the destination that it redirects users to after logging in.